pandas/tests/groupby/test_custom_metadata.py

```diff
@@ -0,0 +1,25 @@
+from pandas import DataFrame
+
+
+class CustomDataFrame(DataFrame):
+    """
+    Extension of DataFrame as described in [guidelines]
+
+    [guidelines]: https://pandas.pydata.org/pandas-docs/stable/development/extending.html#override-constructor-properties  # noqa
+    """
+
+    _metadata = ["_custom_metadata"]
+
+    @property
+    def _constructor(self):
+        return CustomDataFrame
+
+
+def test_groupby_with_custom_metadata():
+    custom_df = CustomDataFrame(
+        [[11, 12, 0], [21, 22, 0], [31, 32, 1]], columns=["a", "b", "g"]
+    )
+    custom_df._custom_metadata = "Custom metadata"
```

Comment:
can you assert that the OP is working as well: https://github.com/pandas-dev/pandas/issues/29442

Reply:
Well spotted -- must admit I did not run the code from the original issue since it was missing inputs. After creating some input and adding the test, I realized that the issue I originally fixed was entirely unrelated...

In any case, I have also proposed a solution for the `.sum()` issue as well. Will resolve this comment, please open a new discussion directly on the proposed solution https://github.com/pandas-dev/pandas/pull/35688/commits/8d3d896bcb20f7bcb0d3ac2b71098dfb05fff11e.

If these two issues should be split into two pull requests, please let me know.